h/traversal/annotation.py

```diff
@@ -26,6 +26,7 @@ class AnnotationContext:
     """Context for annotation-based views."""
 
     annotation: Annotation
+    group: Group
```

Comment:
This is strongly tending towards a global context object which has `user` and all are nullable.

Reply:
How many properties would the global context object have?

I wonder if the `request` object might be the global context:

* If the property you want to add is quite global then add it to the `request`. Examples: `request.authenticated_userid`, `request.identity`, `request.route_url()`, etc. Benefit: write it once and make it available everywhere. Downside: creates global properties.
* If the property you want is more local to one view or route, then add it to the `context`. Benefit: it's only available where it makes sense, to code that uses this context class. Downside: you might have to write it multiple times, or have different views use difference context classes that inherit from the base class, etc, if you're doing this you may be better off with a `request` property?

`request.user` (a user model object) is certainly a common pattern in Pyramid apps. It's partially where the `request.identity` idea in Pyramid 2 came from (and in Pyramid 2 if you wanted to make a user model object available I think you might make that `request.identity` rather than `request.user`?)

Reply:
This is a context object in the sense of a Pyramid context, so no dice there really without changing how Pyramid works. I also think we kind of want to avoid being tangled up with the request (which is much more strongly defined).

The context is always available as `request.context` so `request.context.group` is available etc.

Reply:
The context would have `annotation`, `group`, `organization` and `user`.